File: LeetCode/Intersection of Two Arrays.cpp
//也可先排序，再用两个指针遍历或者用二分查找来实现
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        int len1 = nums1.size();
        int len2 = nums2.size();
        set<int> nums;
        if (len1 <= len2) {
            for (int i = 0; i < len1; i++) {
                for (int j = 0; j < len2; j++) {
                    if (nums1[i] == nums2[j]) {
                        nums.insert(nums1[i]);
                    }
                }
            }
        }
        else {
            for (int i = 0; i < len2; i++) {
                for (int j = 0; j < len1; j++) {
                    if (nums2[i] == nums1[j]) {
                        nums.insert(nums2[i]);
                    }
                }
            }
        }
        vector<int> v;
        set<int>::iterator it;
        for (it = nums.begin(); it != nums.end(); it++) {
            v.push_back(*it);
        }
        return v;
    }
};

File: Lintcode/547_Intersection of Two Arrays/Intersection of Two Arrays.java
public class Solution {
	/**
	 * @param nums1
	 *            an integer array
	 * @param nums2
	 *            an integer array
	 * @return an integer array
	 */
	public int[] intersection(int[] nums1, int[] nums2) {
		Set<Integer> s1 = new HashSet<Integer>();
		Set<Integer> s2 = new HashSet<Integer>();
		for (int i : nums1) {
			s1.add(i);
		}
		for (int i : nums2) {
			if (s1.contains(i)) {
				s2.add(i);
			}
		}
		Object[] o = s2.toArray();
		int[] r = new int[s2.size()];
		for (int i = 0; i < o.length; i++) {
			r[i] = (int) o[i];
		}
		return r;
	}
}